oem/ibm/libpldmresponder/oem_ibm_handler.cpp

```diff
@@ -154,6 +175,106 @@ int pldm::responder::oem_ibm_platform::Handler::
+    pdr->entity_instance = entityInstance;
+    pdr->container_id = 1;
+    pdr->effecter_semantic_id = 0;
+    pdr->effecter_init = PLDM_NO_INIT;
+    pdr->has_description_pdr = false;
+    pdr->composite_effecter_count = 1;
+
+    auto* possibleStatesPtr = pdr->possible_states;
+    auto possibleStates =
+        reinterpret_cast<state_effecter_possible_states*>(possibleStatesPtr);
+    possibleStates->state_set_id = stateSetID;
+    possibleStates->possible_states_size = 2;
+    auto state =
+        reinterpret_cast<state_effecter_possible_states*>(possibleStates);
+    state->states[0].byte = 128;
+    state->states[1].byte = 6;
+    pldm::responder::pdr_utils::PdrEntry pdrEntry{};
+    pdrEntry.data = entry.data();
+    pdrEntry.size = pdrSize;
+    repo.addRecord(pdrEntry);
+}
+
+void attachOemEntityToEntityAssociationPDR(
+    oem_ibm_platform::Handler* platformHandler,
+    pldm_entity_association_tree* bmcEntityTree,
+    const std::string& parentEntityPath, pdr_utils::Repo& repo,
+    [[maybe_unused]] pldm_entity childEntity)
+{
+    auto& associatedEntityMap = platformHandler->getAssociateEntityMap();
+    if (associatedEntityMap.contains(parentEntityPath))
+    {
+        // Parent is present in the entity association PDR
+        pldm_entity parent_entity = associatedEntityMap.at(parentEntityPath);
+        auto parent_node = pldm_entity_association_tree_find_with_locality(
+            bmcEntityTree, &parent_entity, false);
+        if (!parent_node)
+        {
+            // parent node not found in the entity association tree,
+            // this should not be possible
+            error(
+                "Parent Entity of type {ENTITY_TYP} not found in the BMC Entity Association tree ",
+                "ENTITY_TYP", static_cast<unsigned>(parent_entity.entity_type));
+            return;
+        }
+
+        [[maybe_unused]] uint32_t bmc_record_handle = 0;
+
+#ifdef OEM_IBM
+        auto lastLocalRecord = pldm_pdr_find_last_in_range(
+            repo.getPdr(), BMC_PDR_START_RANGE, BMC_PDR_END_RANGE);
+        bmc_record_handle = pldm_pdr_get_record_handle(repo.getPdr(),
+                                                       lastLocalRecord);
+#endif
+
+        /*[[maybe_unused]] uint8_t bmcEventDataOps;
+        uint32_t updatedRecordHdlBmc = 0;
+        bool found = false;
+        pldm_entity_association_find_parent_entity(
+            repo.getPdr(), &parent_entity, false, &updatedRecordHdlBmc, &found);
+        if (found)
+        {
+            pldm_entity_association_pdr_add_contained_entity_to_remote_pdr(
+                repo.getPdr(), &childEntity, updatedRecordHdlBmc);
+            bmcEventDataOps = PLDM_RECORDS_MODIFIED;
+        }
+        else
+        {
+            pldm_entity_association_pdr_create_new(
+                repo.getPdr(), bmc_record_handle, &parent_entity, &childEntity,
+                &updatedRecordHdlBmc);
+            bmcEventDataOps = PLDM_RECORDS_ADDED;
+        }*/
+    }
+}
+
```

Comment:
we should be doing `pldm_entity_association_tree_add()` and `pldm_entity_association_pdr_add_contained_entity()` 


without this this entity won't be present in the entty association PDR.

Reply:
As of now we only need the power off to work to fix the release blocker , this will be added in other PRs

Reply:
Also adding the pldm_entity_association_pdr_add_contained_entity will require another libpldm PR to be reviewed and merged

Reply:
It was decided that we take some time to review and merge the CM libpldm APIs and many reviews are done

Reply:
It is needed to attach this powerStateEntity to the association PDR.
`pldm_entity_association_pdr_add_contained_entity` this is already merged upstream and is in 1110.